browser/brave_vpn/brave_vpn_service_factory.cc

```diff
@@ -77,8 +91,20 @@ KeyedService* BraveVpnServiceFactory::BuildServiceInstanceFor(
       },
       context);
 
-  return new BraveVpnService(shared_url_loader_factory, local_state,
-                             user_prefs::UserPrefs::Get(context), callback);
+  auto* vpn_service =
+      new BraveVpnService(shared_url_loader_factory, local_state,
+                          user_prefs::UserPrefs::Get(context), callback);
+#if BUILDFLAG(IS_WIN)
+  if (base::FeatureList::IsEnabled(
```

Comment:
this is fragile, please move the feature check inside BraveVpnDnsObserverFactory to return nullptr

Reply:
done

Reply:
done? There's still a feature check here that should move

Reply:
removed from here